### ISSUE TITLE
Return the list of errors that cause validity/compatibility issues

### ISSUE THREAD

**Issue by tal181** [Apicurio/apicurio-registry#712]
It would be very helpful if instead of just throwing an exception the APIs will return the cause for the errors.
For example :
if the compatibility rule is Forward.
old schema : 

```
{
  "$id": "https://example.com/person.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "type": "object",
  "properties": {
    "age": {
      "description": "Age in years which must be equal to or greater than zero.",
      "type": "integer",
      "minimum": 0
    }
  }
}
```

```
new schema : {
  "$id": "https://example.com/person.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "type": "object",
  "properties": {
    "age": {
      "description": "Age in years which must be equal to or greater than zero.",
      "type": "string",
      "minimum": 0
    }
  }
}
```

Should return something like "filed age was int but changed to string"

**Comment by adiohana** [comment]
@EricWittmann 
Following PR contains changes for this issue, let me know what you think.
#912

**Comment by adiohana** [comment]
@EricWittmann any estimation when this feature will be in the release version of the Docker image?

**Comment by EricWittmann** [comment]
This functionality is only planned for release as part of the next major release, which will be `2.0.0.Final`.

**Comment by EricWittmann** [comment]
Also this issue is closed with your PR #958 - so closing this now.

**Comment by EricWittmann** [comment]
That said, if you wanted to port the fix to the `1.3.x` branch, I could be persuaded to release `1.3.3.Final` soon.
